File: src/app/models/Item.ts
import { Ingredients } from "./Ingredient";
import { ItemGroup } from "./ItemGroup";
import { Contact } from "./User";

export interface Item {
  id: string
  name: string
  itemGroup: string // salad, sandwitch, etc.
  img: string
  ingredients?: string[]
  description?: string
  price?: string
  ingredientType?: string // itemGroup subtype: eg juice (kind of drink)
  owner?: Contact,
  quantity?: number
}

export type Items = Item[]


export interface CartItem extends Item {
  lastOwner?: Contact
  price: string
  quantity: number
}
export type CartItems = CartItem[]


export interface ItemsByOwner {
  [key: string]: {
    owner: Contact
    items: CartItems,
    total: number,
    isSelected: boolean,
    payStatus: string,
  }
}

File: WebAPI/Models/Item.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebAPI.Data
{
  public class Item
  {
    [Required]
    [Key]
    public string id { get; set; }
    [Required]
    public string name { get; set; }
    public string img { get; set; }
    public List<string> ingredients { get; set; }
    [Required]
    public string itemGroup { get; set; }
    public string description { get; set; }
  }
}
